File: src/main/java/me/wuzzyxy/dynamicmarket/commands/args/HomeCommand.java
package me.wuzzyxy.dynamicmarket.commands.args;

import me.wuzzyxy.dynamicmarket.DynamicMarket;
import me.wuzzyxy.dynamicmarket.market.MarketDatabaseHandler;
import org.bukkit.command.CommandSender;
import java.util.Optional;

public class HomeCommand implements ArgsCommand{
    DynamicMarket plugin;
    private int pushInterval = 0;
    public HomeCommand(DynamicMarket plugin) {
        this.plugin = plugin;
        pushInterval = plugin.getPluginConfig().PUSH_INTERVAL;
    }
    @Override
    public Optional<String[]> execute(String[] args, CommandSender sender) {
        sender.sendMessage("DynamicMarket " + plugin.getDescription().getVersion());

        long currentTime = System.currentTimeMillis();
        long timeSinceLastPush = System.currentTimeMillis() - MarketDatabaseHandler.lastPushTime;
        long nextPushTime = pushInterval*1000L - timeSinceLastPush;

        sender.sendMessage("Push to database in: " + nextPushTime/1000.0 + " s");
        return Optional.empty();
    }


}

File: src/main/kotlin/me/wuzzyxy/dynamicmarket/commands/args/HomeCommand.kt
package me.wuzzyxy.dynamicmarket.commands.args

import me.wuzzyxy.dynamicmarket.DynamicMarket
import org.bukkit.command.CommandSender

class HomeCommand(private val plugin: DynamicMarket) : ArgsCommand {

    override fun execute(args: Array<out String>, sender: CommandSender): List<String>? {
        sender.sendMessage("DynamicMarket ${plugin.description.version}")

        val lastPush = plugin.marketManager.databaseHandler.lastPushTime
        if (lastPush == null) {
            sender.sendMessage("Push to database in: waiting on the first push")
            return null
        }

        // Read live: a reload can change the interval under us.
        val pushInterval = plugin.pluginConfig.PUSH_INTERVAL
        val nextPushTime = pushInterval * 1000L - (System.currentTimeMillis() - lastPush)
        sender.sendMessage("Push to database in: ${nextPushTime / 1000.0} s")
        return null
    }
}
